src/jit/lowerxarch.cpp

```diff
@@ -978,6 +978,75 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerHWIntrinsicCC(node, NI_AVX_PTEST, GenCondition::UGT);
             break;
 
+        case NI_FMA_MultiplyAddScalar: {
+            // Math(F).FusedMultiplyAdd is expanded into NI_FMA_MultiplyAddScalar but
+            // depending on additional GT_NEG nodes it can be:
+            //
+            //    x *  y + z -> NI_FMA_MultiplyAddScalar
+            //    x * -y + z -> NI_FMA_MultiplyAddNegatedScalar
+            //   -x *  y + z -> NI_FMA_MultiplyAddNegatedScalar
+            //   -x * -y + z -> NI_FMA_MultiplyAddScalar
+            //    x *  y - z -> NI_FMA_MultiplySubtractScalar
+            //    x * -y - z -> NI_FMA_MultiplySubtractNegatedScalar
+            //   -x *  y - z -> NI_FMA_MultiplySubtractNegatedScalar
+            //   -x * -y - z -> NI_FMA_MultiplySubtractScalar
+            //
+            GenTreeArgList* argList = node->gtGetOp1()->AsArgList();
+            if (argList->Current()->OperIsHWIntrinsic()) // CreateScalarUnsafe is expected
+            {
+                GenTreeUnOp* createScalarOpX = argList->Current()->AsUnOp();
+                GenTree* argX                = createScalarOpX->gtGetOp1();
+                argList                      = argList->Rest();
+                if (argList->Current()->OperIsHWIntrinsic()) // CreateScalarUnsafe is expected
+                {
+                    GenTreeUnOp* createScalarOpY = argList->Current()->AsUnOp();
+                    GenTree*                argY = createScalarOpY->gtGetOp1();
+                    argList                      = argList->Rest();
+                    if (argList->Current()->OperIsHWIntrinsic()) // CreateScalarUnsafe is expected
+                    {
+                        GenTreeUnOp* createScalarOpZ = argList->Current()->AsUnOp();
+                        GenTree*                argZ = createScalarOpZ->gtGetOp1();
+                        bool                  negMul = argX->OperIs(GT_NEG) ^ argY->OperIs(GT_NEG);
+                        if (argX->OperIs(GT_NEG))
+                        {
+                            GenTree* arg = argX->gtGetOp1();
+
+                            // Drop GT_NEG
+                            // TODO: this doesn't work:
+                            arg->gtNext = createScalarOpX;
+                            createScalarOpX->gtPrev = arg;
```

Comment:
I have here
```
LCL_VAR (arg)
|
NEG (argY)
|
HW_INTRINSIC (createScalarOp2)
```
How can I safely remove GT_NEG from LIR?
I tried just to overwrite `gtPrev` and `gtNext` but it asserts.

Could you please help @tannergooding @mikedn

Reply:
Also, tried different combinations of BlockRange().Insert, etc 🙁 e.g.
```
                            BlockRange().InsertBefore(createScalarOpY, arg);
                            BlockRange().Remove(argY, true);
```

Reply:
> I tried just to overwrite gtPrev and gtNext but it asserts.

You shouldn't need to touch `gtPrev` and `gtNext` directly in LIR, `BlockRange().Remove(x)` is the correct way to do this. What assert are you seeing?

Reply:
@mikedn it complains `found use of a node that is not in the LIR sequence` on GT_NEG

Reply:
@mikedn thanks! it worked